### ISSUE TITLE
Generalization of the of SSP scenarios

### ISSUE THREAD

**Issue by mandresm** [esm-tools/esm_tools#746]
### Discussed in https://github.com/esm-tools/esm_tools/discussions/744

<div type='discussions-op-text'>

<sup>Originally posted by **l975421700** July 22, 2022</sup>
Dear developers

I plan to run an equilibrium SSP585 ECHAM6-wiso standalone simulation on ollie. The first step is to run a standard ssp585 simulation without changing any default boundary conditions. However, I met a problem with it, could you please advise on it? Below are detailed descriptions.

ESM_TOOLS version: 6.11.4
ECHAM version: 6.3.05p2-wiso
yaml scripts: `/work/ollie/qigao001/a_basic_analysis/d_model_simulations/2_echam6/2.2_ssp585/ssp5_echam6_1d.yaml`
Error message while issuing esm_runscripts. command:

> qigao001> esm_runscripts /work/ollie/qigao001/a_basic_analysis/d_model_simulations/2_echam6/2.2_ssp585/ssp5_echam6_1d.yaml -e test
> Copying standard yamls from:  /home/ollie/qigao001/esm_tools/configs/
> Copying standard namelists from:  /home/ollie/qigao001/esm_tools/namelists/
> /work/ollie/qigao001/a_basic_analysis/d_model_simulations/2_echam6/2.2_ssp585/ssp5_echam6_1d.yaml
> First year, checking if we need to compile...
> 
> STARTING SIMULATION JOB!
> Experiment ID = test
> Setup = echam
> Experiment is installed in:
>        /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5//test
> 
> 
> ERROR: Missing Configuration
> The input file variable LU_HIST of forcing_sources can not be fully resolved:
> 
> /work/ollie/pool/JSBACH/input/r0010/T63/New_Hampshire_LCC/hist_harvest/LUH_harvest_T63_1850.nc:
>     to: 1850
> /work/ollie/pool/JSBACH/input/r0010/T63/New_Hampshire_LCC/hist_harvest/LUH_harvest_T63_@YEAR@.nc:
>     from: 1851

There is a similar problem in another ticket: https://github.com/esm-tools/esm_tools/discussions/673. But I cannot find an answer there.

That's for a different model but with the same error message. The suggested link there does not work anymore.

Regards, Qinggang</div>

**Comment by mandresm** [comment]
Hi Qinggang, I think you should be able to run it both in Levante and Ollie, it's up totally up to you.

About the error, there are different components that need to be solved for this to work:

1. **ESM-Tools version**: You are running year 2015, and that version of ESM-Tools tries to copy the following file into the experiment and that file does not exist: 
```
/work/ollie/pool/JSBACH/input/r0010/T63/New_Hampshire_LCC/hist_harvest/LUH_harvest_T63_2015.nc
```
In that version of ESM-Tools an ssp585 scenario was simply a historic cmip6 scenario, so it had not the capacity of handling ssp scenarios further away from 2014, the year where `hist_harvest` files finish:
https://github.com/esm-tools/esm_tools/blob/ed7b25a1aefdc3385ca59269716b84a91095b8a5/configs/components/jsbach/jsbach.datasets.yaml#L15-L19
As you can see here, from year 1851 it will always try to load from `hist_harvest` folder, which does not contain the "ssp years". We solved this recently (version 6.13.1), now for ssp scenarios the files loaded for harvest follow this rule:
https://github.com/esm-tools/esm_tools/blob/067711bcb4e6483135270b2023a8a45bc3137ee9/configs/components/jsbach/jsbach.datasets.yaml#L115-L122
so from 2015 onwards it stops taking the files from the `hist_harvest` folder and starts to get them from the `${echam.scenario}_harvest` folder, in your case `ssp585_harvest`. That leads us to the next problem:
2. **Pools in Ollie not complete**: the ssp folders for JSBACH are not present in Ollie. If you do `ls` in the corresponding Levante pool:
```
$ ls /pool/data/JSBACH/input/r0010/T63/New_Hampshire_LCC/
hist_dynveg        hist-high_no-dynveg  hist_no-dynveg                              ssp119_dynveg     ssp126_harvest    ssp245_no-dynveg  ssp434_dynveg     ssp460_harvest    ssp534os_no-dynveg
hist_harvest       hist-low_dynveg      LUH_transitions_T63_all-crop-to-natural.nc  ssp119_harvest    ssp126_no-dynveg  ssp370_dynveg     ssp434_harvest    ssp460_no-dynveg  ssp585_dynveg
hist-high_dynveg   hist-low_harvest     LUH_transitions_T63_all-pasture-to-crop.nc  ssp119_no-dynveg  ssp245_dynveg     ssp370_harvest    ssp434_no-dynveg  ssp534os_dynveg   ssp585_harvest
hist-high_harvest  hist-low_no-dynveg   no_LUH_transitions_T63.nc                   ssp126_dynveg     ssp245_harvest    ssp370_no-dynveg  ssp460_dynveg     ssp534os_harvest  ssp585_no-dynveg
```
In Ollie:
```
$ ls /work/ollie/pool/JSBACH/input/r0010/T63/New_Hampshire_LCC/
hist_dynveg  hist_harvest  hist_no-dynveg  LUH_transitions_T63_all-crop-to-natural.nc  LUH_transitions_T63_all-pasture-to-crop.nc  no_LUH_transitions_T63.nc
```
@pgierz, can we update the JSBACH pools in Ollie?

**In summary**
Please wait for someone to update the pools in Ollie. I will also need to produce a minor fix to the ssp scenarios. In addition to that there were some adaptations for ssp scenarios recently made for AWICM1-REcoM and Levante, but those have not fully transferred to Ollie and standalone ECHAM. This is a good chance to fix the problem so I'm turning this into an issue and we will stay in contact through out the week to be solving the problems and offering full support to ssp scenarios in both machines and models.

**Comment by QinggangGao** [comment]
Thanks a lot, Miguel. Looking forward to the solution and I will be happy to test it.

**Comment by pgierz** [comment]
Copying from Levante to Ollie is finished; please test if it all works now and let me know.

**Comment by mandresm** [comment]
Thanks!

**Comment by mandresm** [comment]
I'm afraid the pool issue is not yet solved because I did not report the whole list of what it's missing (sorry about that). In summary it seems that the ollie pools for ECHAM and JSBACH are not up to date:
```
WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/greenhouse_ssp585.nc
WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/MAC-SP_ssp585.nc
WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/ozone/T63_ozone_ssp585_2015.nc
WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/ozone/T63_ozone_ssp585_2016.nc
WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/ozone/T63_ozone_ssp585_2017.nc
WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_sw_T63_2015.nc
WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_sw_T63_2016.nc
WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_sw_T63_2017.nc
WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_ir_T63_2015.nc
WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_ir_T63_2016.nc
WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_ir_T63_2017.nc
WARNING: File not found: /work/ollie/pool/JSBACH/input/r0010/T63/land_use_maps/ssp585/cover_fract_T63_11tiles_2015.nc
WARNING: File not found: /work/ollie/pool/JSBACH/input/r0010/T63/New_Hampshire_LCC/ssp585_no_dynveg/LUH_scaled_transitions_T63_2015.nc
WARNING: File not found: /work/ollie/pool/JSBACH/input/r0010/T63/New_Hampshire_LCC/hist_no_dynveg/LUH_scaled_transitions_T63_2014.nc
WARNING: File not found: /work/ollie/pool/JSBACH/input/r0010/T63/ndepo/Ndepo_ssp585.2015.nc
WARNING: File not found: /work/ollie/pool/JSBACH/input/r0010/T63/ndepo/Ndepo.2014.nc
```

**Comment by pgierz** [comment]
Time for more `rsync` then....and time to take a day and make this all automatic. I'll post here once all the copying is finished.

**Comment by mandresm** [comment]
Thanks a lot Paul!

**Comment by QinggangGao** [comment]
Thanks both Paul and Miguel

I met another problem, which might also connect to the update of pool. Can you please advise?

ECHAM6 version: 6.3.05p2-wiso
esm_tools version: 6.11.4
scenario: "PI-CTRL"

I run a simulation for 27 years here: `/work/ollie/qigao001/output/echam-6.3.05p2-wiso/pi/pi_m_411_4.9`.
But when I restart it two days ago from its own scripts folder`/work/ollie/qigao001/output/echam-6.3.05p2-wiso/pi/pi_m_411_4.9/scripts/pi_echam6_1m.yaml`, the log file indicates FileNotFoundError.

The log file is here: `/work/ollie/qigao001/output/echam-6.3.05p2-wiso/pi/pi_m_411_4.9/run_20271001-20271031/log/pi_m_411_4.9_echam_compute_16576349.log`. And the error is copied below:

FileNotFoundError: [Errno 2] No such file or directory: '/work/ollie/qigao001/output/echam-6.3.05p2-wiso/pi//pi_m_411_4.9/run_20271001-20271031/forcing/jsbach/landuseHarvest.2027.nc'

Regards, Qinggang

**Comment by QinggangGao** [comment]
Hi developers

May I ask is there any progress on this issue? If you need somebody to test, I am very happy to do it.

The above FileNotFoundError regarding 'landuseHarvest.*.nc' does not appear anymore.

Regards, Qinggang

**Comment by pgierz** [comment]
Hi Qinggang,

which files are now still missing? I updated everything yesterday, so you should be good to go. Let me know if anything is not yet in the pool so I can take care of it.

Best
PG

**Comment by mandresm** [comment]
The generalization of the SSP scenarios is still missing. I put together a branch a week ago to solve the problem (#747): `fix/ssp_generalization_jsbach`. Qinggang, can you checkout that branch of ESM-Tools and give it a try? You'll need to specify:
```
echam:
    scenario: ssp<3-digit-number>
```

**Comment by QinggangGao** [comment]
Thanks. I tried but failed. A list of files are missing.

I checked out the esm_tools branch `fix/ssp_generalization_jsbach `.
The yaml script is here: `/work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/scripts/ssp5_echam6_1d.yaml`.
The log file is here: `/work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/log/test_ssp585_echam_compute_20150101-20150131_16757820.log`.

Main error messages are listed below:

>   0:  Read ozone climatology from unit 21
>   0:  WARNING: read_o3clim_3: you chose zonal mean ozone boundary conditions. No transient boundary condition is possible
>   0:  IO_open failed on: unit.21
>   0:  IO_open: No such file or directory
>   0: 
>   0: ================================================================================
>   0: 
>   0: FATAL ERROR in IO_open: Run terminated.
>   0:  FINISH called from PE: 0

Missing files are reported below:

> esm_tools> esm_runscripts /work/ollie/qigao001/a_basic_analysis/d_model_simulations/2_echam6/2.2_ssp585/ssp5_echam6_1d.yaml -e test_ssp585 -c
> Copying standard yamls from:  /home/ollie/qigao001/esm_tools/configs/
> Copying standard namelists from:  /home/ollie/qigao001/esm_tools/namelists/
> /work/ollie/qigao001/a_basic_analysis/d_model_simulations/2_echam6/2.2_ssp585/ssp5_echam6_1d.yaml
> First year, checking if we need to compile...
> 
> ================================================================================
> STARTING SIMULATION JOB!
> Experiment ID = test_ssp585
> Setup = echam
> Experiment is installed in:
>        /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5//test_ssp585
> ================================================================================
> 
> WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/greenhouse_ssp585.nc
> WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/MAC-SP_ssp585.nc
> WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/ozone/T63_ozone_ssp585_2015.nc
> WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/ozone/T63_ozone_ssp585_2016.nc
> WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/ozone/T63_ozone_ssp585_2017.nc
> WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_sw_T63_2015.nc
> WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_sw_T63_2016.nc
> WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_sw_T63_2017.nc
> WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_ir_T63_2015.nc
> WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_ir_T63_2016.nc
> WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_ir_T63_2017.nc
> WARNING: File not found: /work/ollie/pool/JSBACH/input/r0010/T63/land_use_maps/ssp585/cover_fract_T63_11tiles_2015.nc
> WARNING: File not found: /work/ollie/pool/JSBACH/input/r0010/T63/New_Hampshire_LCC/ssp585_no_dynveg/LUH_scaled_transitions_T63_2015.nc
> WARNING: File not found: /work/ollie/pool/JSBACH/input/r0010/T63/New_Hampshire_LCC/hist_no_dynveg/LUH_scaled_transitions_T63_2014.nc
> WARNING: File not found: /work/ollie/pool/JSBACH/input/r0010/T63/ndepo/Ndepo_ssp585.2015.nc
> WARNING: File not found: /work/ollie/pool/JSBACH/input/r0010/T63/ndepo/Ndepo.2014.nc
> WARNING: File not found: /home/ollie/qigao001/esm_tools/namelists/echam/6.3.05p2-wiso/cmip6/namelist.echam
> WARNING: File not found: /home/ollie/qigao001/esm_tools/namelists/jsbach/3.20/ssp585/no_dynveg/namelist.jsbach
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/greenhouse_gases.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/MAC-SP.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/ozon2015
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/ozon2016
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/ozon2017
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_sw_2015.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_sw_2016.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_sw_2017.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_ir_2015.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_ir_2016.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_ir_2017.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/jsbach/cover_fract.2015.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/jsbach/landuseTransitions.2015.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/jsbach/landuseTransitions.2014.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/jsbach/Ndepo.2015.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/jsbach/Ndepo.2014.nc
> MISSING FILES:
> 
> - missing source: /work/ollie/pool/ECHAM6/input/r0007/greenhouse_ssp585.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/greenhouse_gases.nc
> 
> - missing source: /work/ollie/pool/ECHAM6/input/r0007/MAC-SP_ssp585.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/MAC-SP.nc
> 
> - missing source: /work/ollie/pool/ECHAM6/input/r0007/T63/ozone/T63_ozone_ssp585_2015.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/ozon2015
> 
> - missing source: /work/ollie/pool/ECHAM6/input/r0007/T63/ozone/T63_ozone_ssp585_2016.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/ozon2016
> 
> - missing source: /work/ollie/pool/ECHAM6/input/r0007/T63/ozone/T63_ozone_ssp585_2017.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/ozon2017
> 
> - missing source: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_sw_T63_2015.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_sw_2015.nc
> 
> - missing source: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_sw_T63_2016.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_sw_2016.nc
> 
> - missing source: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_sw_T63_2017.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_sw_2017.nc
> 
> - missing source: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_ir_T63_2015.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_ir_2015.nc
> 
> - missing source: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_ir_T63_2016.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_ir_2016.nc
> 
> - missing source: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_ir_T63_2017.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_ir_2017.nc
> 
> - missing source: /work/ollie/pool/JSBACH/input/r0010/T63/land_use_maps/ssp585/cover_fract_T63_11tiles_2015.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/jsbach/cover_fract.2015.nc
> 
> - missing source: /work/ollie/pool/JSBACH/input/r0010/T63/New_Hampshire_LCC/ssp585_no_dynveg/LUH_scaled_transitions_T63_2015.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/jsbach/landuseTransitions.2015.nc
> 
> - missing source: /work/ollie/pool/JSBACH/input/r0010/T63/New_Hampshire_LCC/hist_no_dynveg/LUH_scaled_transitions_T63_2014.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/jsbach/landuseTransitions.2014.nc
> 
> - missing source: /work/ollie/pool/JSBACH/input/r0010/T63/ndepo/Ndepo_ssp585.2015.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/jsbach/Ndepo.2015.nc
> 
> - missing source: /work/ollie/pool/JSBACH/input/r0010/T63/ndepo/Ndepo.2014.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/jsbach/Ndepo.2014.nc
> 
> - missing source: /home/ollie/qigao001/esm_tools/namelists/echam/6.3.05p2-wiso/cmip6/namelist.echam
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/config/echam/namelist.echam
> 
> - missing source: /home/ollie/qigao001/esm_tools/namelists/jsbach/3.20/ssp585/no_dynveg/namelist.jsbach
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/config/jsbach/namelist.jsbach
> 
> - missing source: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/greenhouse_gases.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/work/greenhouse_gases.nc
> 
> - missing source: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/MAC-SP.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/work/MAC-SP.nc
> 
> - missing source: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/ozon2015
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/work/ozon2015
> 
> - missing source: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/ozon2016
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/work/ozon2016
> 
> - missing source: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/ozon2017
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/work/ozon2017
> 
> - missing source: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_sw_2015.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/work/strat_aerosol_sw_2015.nc
> 
> - missing source: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_sw_2016.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/work/strat_aerosol_sw_2016.nc
> 
> - missing source: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_sw_2017.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/work/strat_aerosol_sw_2017.nc
> 
> - missing source: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_ir_2015.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/work/strat_aerosol_ir_2015.nc
> 
> - missing source: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_ir_2016.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/work/strat_aerosol_ir_2016.nc
> 
> - missing source: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_ir_2017.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/work/strat_aerosol_ir_2017.nc
> 
> - missing source: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/jsbach/cover_fract.2015.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/work/cover_fract.2015.nc
> 
> - missing source: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/jsbach/landuseTransitions.2015.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/work/landuseTransitions.2015.nc
> 
> - missing source: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/jsbach/landuseTransitions.2014.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/work/landuseTransitions.2014.nc
> 
> - missing source: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/jsbach/Ndepo.2015.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/work/Ndepo.2015.nc
> 
> - missing source: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/jsbach/Ndepo.2014.nc
> - missing target: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/work/Ndepo.2014.nc
> ================================================================================
> 
> tasks: 576
> resubmit tasks: 576
> Actually not submitting anything, this job preparation was launched in 'check' mode (-c).

**Comment by mandresm** [comment]
Try:
```
jsbach:
    dataset: r0010
```

**Comment by QinggangGao** [comment]
Same error happened again:

> WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/greenhouse_ssp585.nc
> WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/MAC-SP_ssp585.nc
> WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/ozone/T63_ozone_ssp585_2015.nc
> WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/ozone/T63_ozone_ssp585_2016.nc
> WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/ozone/T63_ozone_ssp585_2017.nc
> WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_sw_T63_2015.nc
> WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_sw_T63_2016.nc
> WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_sw_T63_2017.nc
> WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_ir_T63_2015.nc
> WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_ir_T63_2016.nc
> WARNING: File not found: /work/ollie/pool/ECHAM6/input/r0007/T63/volcano_aerosols/strat_aerosol_ir_T63_2017.nc
> WARNING: File not found: /work/ollie/pool/JSBACH/input/r0010/T63/land_use_maps/ssp585/cover_fract_T63_11tiles_2015.nc
> WARNING: File not found: /work/ollie/pool/JSBACH/input/r0010/T63/New_Hampshire_LCC/ssp585_no_dynveg/LUH_scaled_transitions_T63_2015.nc
> WARNING: File not found: /work/ollie/pool/JSBACH/input/r0010/T63/New_Hampshire_LCC/hist_no_dynveg/LUH_scaled_transitions_T63_2014.nc
> WARNING: File not found: /work/ollie/pool/JSBACH/input/r0010/T63/ndepo/Ndepo_ssp585.2015.nc
> WARNING: File not found: /work/ollie/pool/JSBACH/input/r0010/T63/ndepo/Ndepo.2014.nc
> WARNING: File not found: /home/ollie/qigao001/esm_tools/namelists/echam/6.3.05p2-wiso/cmip6/namelist.echam
> WARNING: File not found: /home/ollie/qigao001/esm_tools/namelists/jsbach/3.20/ssp585/no_dynveg/namelist.jsbach
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/greenhouse_gases.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/MAC-SP.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/ozon2015
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/ozon2016
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/ozon2017
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_sw_2015.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_sw_2016.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_sw_2017.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_ir_2015.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_ir_2016.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/echam/strat_aerosol_ir_2017.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/jsbach/cover_fract.2015.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/jsbach/landuseTransitions.2015.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/jsbach/landuseTransitions.2014.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/jsbach/Ndepo.2015.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test_ssp585/run_20150101-20150131/forcing/jsbach/Ndepo.2014.nc
> MISSING FILES:
>

**Comment by mandresm** [comment]
Okay, so part of the problem would be solved if the ECHAM r0008 is updated with what it's defined in Levante. @pgierz , can you do that? That would be the folder in levante: `/pool/data/ECHAM6/input/r0008/` (r0008 does not exist in Ollie)

Once we get confirmatuion by @pgierz , you can try the following:
```
echam:
    dataset: r0008
```

You'll still meet some missing files but they should be less, and then we can continue with the following problem.

**Comment by QinggangGao** [comment]
Good, thanks

**Comment by QinggangGao** [comment]
Hi developers

Hope you have a nice summer and holidays.

I tried again. The missing files seem to be less than previous. But I still cannot run it.

esm_tools, version 6.13.1, branch: fix/ssp_generalization_jsbach
 
experiment on ollie: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test

esm_runscripts warning message:

> esm_tools> esm_runscripts ssp5_echam6_1d.yaml -e test
> Copying standard yamls from:  /home/ollie/qigao001/esm_tools/configs/
> Copying standard namelists from:  /home/ollie/qigao001/esm_tools/namelists/
> ssp5_echam6_1d.yaml
> First year, checking if we need to compile...
> 
> 
> STARTING SIMULATION JOB!
> Experiment ID = test
> Setup = echam
> Experiment is installed in:
>        /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5//test
> 
> 
> WARNING: File not found: /work/ollie/pool/JSBACH/input/r0010/T63/land_use_maps/ssp585/cover_fract_T63_11tiles_2015.nc
> WARNING: File not found: /work/ollie/pool/JSBACH/input/r0010/T63/New_Hampshire_LCC/ssp585_no_dynveg/LUH_scaled_transitions_T63_2015.nc
> WARNING: File not found: /work/ollie/pool/JSBACH/input/r0010/T63/New_Hampshire_LCC/hist_no_dynveg/LUH_scaled_transitions_T63_2014.nc
> WARNING: File not found: /work/ollie/pool/JSBACH/input/r0010/T63/ndepo/Ndepo_ssp585.2015.nc
> WARNING: File not found: /work/ollie/pool/JSBACH/input/r0010/T63/ndepo/Ndepo.2014.nc
> WARNING: File not found: /home/ollie/qigao001/esm_tools/namelists/echam/6.3.05p2-wiso/cmip6/namelist.echam
> WARNING: File not found: /home/ollie/qigao001/esm_tools/namelists/jsbach/3.20/ssp585/no_dynveg/namelist.jsbach
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test/run_20150101-20150131/forcing/jsbach/cover_fract.2015.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test/run_20150101-20150131/forcing/jsbach/landuseTransitions.2015.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test/run_20150101-20150131/forcing/jsbach/landuseTransitions.2014.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test/run_20150101-20150131/forcing/jsbach/Ndepo.2015.nc
> WARNING: File not found: /work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test/run_20150101-20150131/forcing/jsbach/Ndepo.2014.nc

log file error message: 

> Module for impi,  version 2018.4.274 loaded.
> 252: slurmstepd: error: couldn't chdir to `/work/ollie/qigao001/output/echam-6.3.05p2-wiso/ssp5/test/run_20150101-20150131/work': No such file or directory: going to /tmp instead
> 252: slurmstepd: error: Unable to create TMPDIR [/work/ollie/tmp]: Permission denied
> 252: slurmstepd: error: Setting TMPDIR to /tmp
> 257: slurmstepd: error: execve(): /tmp/./echam6_0720_4.9: No such file or directory
>

**Comment by mandresm** [comment]
Hi Qinggang, I just came back from holidays, I will look at this today. The problem is some missing files in Ollie, this time, not the standard pool, but some files that need to be preprocessed from the pool, for ssp simulations. These files already exist on Levante, so I will make sure they get copied into Ollie. Apologies for the delay on the resolution.

**Comment by QinggangGao** [comment]
Hi Miguel

Thanks a lot. Pengyang posted a solution here: https://github.com/esm-tools/esm_tools/discussions/673.

I will try it later, could you please also have a check?

Regards, Qinggang

**Comment by mandresm** [comment]
Dear Qinggang, I think that solution is misleading: the model will run and no files will be reported missing by ESM-Tools but you'll be using Ndepo files of 1850 for every run of your simulation, no matter what year it is currently in. I cannot recommend this approach. Instead, PR #747 should solve that, once finishes. @pgierz just copied the necessary ssp Ndepo files from Levante to Ollie pool, and I can continue with that PR. I'll ask you for testing that branch ASAP (before the end of the week).

**Comment by QinggangGao** [comment]
Thanks. I'd be happy to test.

**Comment by nwieters** [comment]
Hi @l975421700,

can you please give an update on this issue? Since ollie is not running any more, have you been able to run the SSP scenarios on another HPC, like Levante or Albedo?

Thank you for your update on this.

**Comment by QinggangGao** [comment]
Hi 

TBH I lost the track a while ago and never picked it up again. I might be interested in running an ssp585 simulation using ECHAM6 on albedo in the next two months, but not for now. This issue can be closed for me. In case of any problems on albedo, I will report again.

**Comment by nwieters** [comment]
Thank you @l975421700 for your reply. It would be very interesting to know how your SSP simulations run on Albedo.
